pallets/medical-record/src/tests.rs

```diff
@@ -45,20 +49,53 @@ fn patient_can_add_record() {
 				})
 				.collect::<Vec<_>>()
 				.len(),
-			max_record_len as usize
+			max_record_len
 		);
 
 		assert_noop!(
 			MedicalRecord::patient_adds_record(o.clone(), BoundedVec::with_max_capacity()),
 			Error::<Test>::ExceedsMaxRecordLength
 		);
+	});
+}
+
+#[test]
+fn doctor_can_add_record_for_patient() {
+	new_test_ext().execute_with(|| {
+		let doctor = RuntimeOrigin::signed(1);
+		let patient = RuntimeOrigin::signed(2);
+		assert_ok!(MedicalRecord::create_account(doctor.clone(), UserType::Doctor));
+		assert_ok!(MedicalRecord::create_account(patient.clone(), UserType::Patient));
+
+		let max_record_len = <MockMaxRecordLength as Get<u32>>::get() as usize;
+		for _ in 0..max_record_len {
+			assert_ok!(MedicalRecord::doctor_adds_record(
+				doctor.clone(),
+				origin_to_account_id(patient.clone()),
+				BoundedVec::with_max_capacity(),
+				BoundedVec::with_max_capacity(),
+			));
+		}
+
+		let patient_records =
+			MedicalRecord::records(origin_to_account_id(patient.clone()), UserType::Patient)
+				.unwrap();
```

Comment:
[Nitpicking]
Could you please change the unwrap() to a expect() just for the sake of clarity?